torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -231,23 +233,82 @@ class ValidateAllocation : private kir::IrVisitor {
   std::vector<std::vector<const Allocate*>> live_allocations_;
 };
 
+class MemoizeAllocations : kir::IrVisitor {
+ public:
+  static void memoizeAllocations(Kernel* kernel, ExprBuilderVisitor& visitor) {
+    MemoizeAllocations(visitor, kernel, kernel->summary());
+  }
+
+ private:
+  MemoizeAllocations(
+      ExprBuilderVisitor& builder_visitor,
+      const Kernel* kernel,
+      KernelSummary& summary)
+      : summary_(&summary) {
+    TORCH_CHECK(summary_->memoized_dag == nullptr);
+    builder_visitor_ =
+        &builder_visitor; //.reset(new ExprBuilderVisitor(builder_, *kernel));
+
+    for (const auto& ir_node : kernel->irNodes())
+      ir_node->accept(this);
+  }
+
+  void memoizeGlobalAllocationShape(const kir::Allocate* allocate) {
+    std::vector<glfdc::ExprEvaluator> shapes;
+
+    for (auto shape_expr : allocate->shape()) {
+      auto expr = builder_visitor_->build(shape_expr);
+      TORCH_CHECK(expr.has_value());
+      shapes.emplace_back(expr.value());
+    }
+    summary_->memoized_global_shapes.push_back(shapes);
+  }
+
+  void visit(const kir::Allocate* allocate) final {
+    switch (allocate->memoryType()) {
+      case MemoryType::Global: {
+        auto expr = builder_visitor_->build(allocate->size());
+        if (expr.has_value())
+          summary_->memoized_global_allocations.emplace_back(expr.value());
```

Comment:
Do we need this? `memoized_global_allocations` doesn't seem to be used.

Reply:
It's not used, will remove it

Reply:
Removed in current version